python_modules/dagster/dagster_tests/daemon_tests/test_backfill.py

```diff
@@ -574,7 +578,25 @@ def test_simple_backfill(
     )
     assert instance.get_runs_count() == 0
 
-    list(execute_backfill_iteration(workspace_context, get_default_daemon_logger("BackfillDaemon")))
+    if parallel:
+        backfill_daemon_futures = {}
+        list(
+            execute_backfill_iteration(
+                workspace_context,
+                get_default_daemon_logger("BackfillDaemon"),
+                threadpool_executor=ThreadPoolExecutor(2),
+                backfill_futures=backfill_daemon_futures,
+            )
+        )
+
+        wait_for_futures(backfill_daemon_futures)
+        [list(future.result()) for future in backfill_daemon_futures.values()]
```

Comment:
@gibsondan I still feel this sort of logic is incorrect.

Reading the `wait_for_futures` implementation, I would think something like the following should work:

```suggestion
        [list(result) for result in wait_for_futures(backfill_daemon_futures).values()]
```

But it doesn't, and so it seems the second line after `wait_for_futures` in the current code _IS_ doesn't something...

I feel like I'm misunderstanding how `wait_for_futures` should be used, and that this comes to a head in the issue with `test_two_backfills_at_the_same_time`.

Reply:
@OwenKephart helped debug `test_two_backfills_at_the_same_time` in particular for a while, and a couple observations that came out of it:

* It does seem the `[list(future.result()) for future in backfill_daemon_futures.values()]` line is doing everything we might need from `wait_for_futures(backfill_daemon_futures)`, especially since we're not doing any fancy `timeout`-driven behavior.
* Removing the `wait_for_futures()` call fixes the `test_two_backfills_at_the_same_time` issue?

So I've gone ahead and removed all the `wait_for_futures()` calls that I'd added initially.

Reply:
Previous comment was wrong; after trying to get https://github.com/dagster-io/dagster/pull/26189/files#discussion_r1887135628 working, realized that the problem is the backfill iterations were not being run in thread. After fixing that, `wait_for_futures()` works as expected!

Unfortunately, there seems to be some (deadlock?) issue with running two backfills at once now with the `ThreadPoolExecutor`, which I still need to figure out...